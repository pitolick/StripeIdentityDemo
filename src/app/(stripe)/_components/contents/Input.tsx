/**
 * Stripeフォーム個人情報入力画面の内容を定義するコンポーネント
 * @module StripeInputContents
 */
import React from 'react'
import { HeadingCaption } from '@/components/elements'
import { StripeElements, StripePaymentIntent, StripeIdentityVerificationSession } from '@/components/plugins/Stripe'
import { useFormContext } from 'react-hook-form'

/**
 * Stripeフォーム個人情報入力画面の内容を定義するProps
 * @typedef {Object} StripeInputContentsProps
 */
export interface StripeInputContentsProps {}

/**
 * Stripeフォーム個人情報入力画面の内容を定義するコンポーネント
 * @param {StripeInputContentsProps} props - Stripeフォーム個人情報入力画面の内容を定義するコンポーネントのプロパティ
 * @returns {JSX.Element} - Stripeフォーム個人情報入力画面の内容を定義するコンポーネント
 */
export const StripeInputContents: React.FC<StripeInputContentsProps> = (props) => {
  const {
    formState: { errors },
  } = useFormContext()

  /**
   * フォームの設定
   */

  return (
    <>
      {/* お支払い情報 */}
      <div className="mt-16">
        <HeadingCaption className="mb-8 text-lg md:text-xl">お支払い情報</HeadingCaption>
        <div className="space-y-5">
          {/* StripeElements */}
          <StripeElements intentType="setup">
            <StripePaymentIntent nameId="stripeSetupIntent" label="カード情報" required={true} error={errors.stripeSetupIntent} />
          </StripeElements>
        </div>
      </div>

      {/* 本人確認 */}
      <div className="mt-16">
        <HeadingCaption className="mb-8 text-lg md:text-xl">本人確認</HeadingCaption>
        <div className="space-y-5">
          <StripeIdentityVerificationSession nameId="stripeIdentity" label="本人確認" required={true} error={errors.stripeIdentity} />
        </div>
      </div>
    </>
  )
}
