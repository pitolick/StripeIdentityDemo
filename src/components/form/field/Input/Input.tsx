/**
 * 入力項目非制御コンポーネント
 * @module Input
 */
import React, { forwardRef } from 'react'

/**
 * 入力項目コンポーネントのプロパティ
 * @typedef {Object} InputProps
 * @property {React.InputHTMLAttributes<HTMLInputElement>} - 入力項目タグのHTML属性
 */
export interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {}

/**
 * 入力項目非制御コンポーネント
 * @param {InputProps} props - 入力項目コンポーネントのプロパティ
 * @returns {JSX.Element} - 入力項目コンポーネント
 */
export const Input = forwardRef<HTMLInputElement, InputProps>((props, ref) => {
  const { className = undefined, ...inputProps } = props
  const classNameProps = className ? className : ''
  return (
    <>
      <input
        ref={ref}
        {...inputProps}
        className={`${classNameProps} block w-full rounded-md border border-gray-500 px-3 py-2 text-sm file:mr-4 file:border-0 file:bg-gray-300 file:px-4 file:py-3 md:px-5 md:py-3 md:text-base`}
      />
    </>
  )
})

Input.displayName = 'Input'
