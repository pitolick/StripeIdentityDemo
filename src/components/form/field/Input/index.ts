export * from './Input'
export * from './InputControl'
