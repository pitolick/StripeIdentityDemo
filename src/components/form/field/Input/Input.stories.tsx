import type { Meta, StoryObj } from '@storybook/react'
import { Input } from './Input'

const meta: Meta<typeof Input> = {
  title: 'Form/Field/Input/Base',
  component: Input,
  tags: ['autodocs'],
}

export default meta
type Story = StoryObj<typeof Input>

/**
 * 基本形：テキスト
 */
export const Base: Story = {
  args: {
    type: 'text',
    placeholder: '入力してください',
  },
}

/**
 * 基本形：パスワード
 */
export const Password: Story = {
  args: {
    type: 'password',
    placeholder: 'パスワードを入力してください',
  },
}

/**
 * 基本形：日付
 */
export const Date: Story = {
  args: {
    type: 'date',
    placeholder: '日付を選択してください',
  },
}

/**
 * 基本形：時刻
 */
export const Time: Story = {
  args: {
    type: 'time',
    placeholder: '時刻を選択してください',
  },
}

/**
 * 基本形：日時
 */
export const DateTime: Story = {
  args: {
    type: 'datetime-local',
    placeholder: '日時を選択してください',
  },
}

/**
 * 基本形：月
 */
export const Month: Story = {
  args: {
    type: 'month',
    placeholder: '月を選択してください',
  },
}

/**
 * 基本形：週
 */
export const Week: Story = {
  args: {
    type: 'week',
    placeholder: '週を選択してください',
  },
}

/**
 * 基本形：メールアドレス
 */
export const Email: Story = {
  args: {
    type: 'email',
    placeholder: 'メールアドレスを入力してください',
  },
}

/**
 * 基本形：電話番号
 */
export const Tel: Story = {
  args: {
    type: 'tel',
    placeholder: '電話番号を入力してください',
  },
}

/**
 * 基本形：URL
 */
export const Url: Story = {
  args: {
    type: 'url',
    placeholder: 'URLを入力してください',
  },
}

/**
 * 基本形：検索
 */
export const Search: Story = {
  args: {
    type: 'search',
    placeholder: '検索してください',
  },
}

/**
 * 基本形：数値
 */
export const Number: Story = {
  args: {
    type: 'number',
    placeholder: '数値を入力してください',
  },
}

/**
 * 基本形：範囲
 */
export const Range: Story = {
  args: {
    type: 'range',
    placeholder: '範囲を選択してください',
  },
}

/**
 * 基本形：ファイル
 */
export const File: Story = {
  args: {
    type: 'file',
    placeholder: 'ファイルを選択してください',
  },
}

/**
 * 基本形：非表示
 */
export const Hidden: Story = {
  args: {
    type: 'hidden',
    placeholder: '非表示を選択してください',
  },
}
