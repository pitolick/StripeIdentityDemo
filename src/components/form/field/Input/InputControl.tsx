/**
 * 入力項目制御コンポーネント（react-hook-form）
 * @module InputControl
 */
import React, { forwardRef } from 'react'
import { Input, InputProps } from './Input'
import { InputBox, InputBoxProps } from '@/components/form/parts'

/**
 * 入力項目コンポーネントのプロパティ
 * @memo バリデーションスキーマ（zod）は各入力子コンポーネント内でexportしformコンポーネント内でresolverに指定する
 * @typedef {Object} InputControlProps
 */

export interface InputControlProps extends InputProps, Omit<InputBoxProps, 'children'> {}

/**
 * 入力項目制御コンポーネント（react-hook-form）
 * @param {InputControlProps} props - 入力項目コンポーネントのプロパティ
 * @returns {JSX.Element} - 入力項目コンポーネント
 */
export const InputControl = forwardRef<HTMLInputElement, InputControlProps>((props, ref) => {
  const { nameId, label, note, required = undefined, className, error, afterChildren, contentWidth, ...inputProps } = props
  const classNameProps = className ? className : ''
  // エラー時のclass名
  const errorClass = error ? `bg-red-pale outline outline-red outline-2 -outline-offset-2` : ''
  return (
    <InputBox nameId={nameId} label={label} note={note} required={required} error={error} afterChildren={afterChildren} contentWidth={contentWidth}>
      {/* 入力項目 */}
      <Input id={nameId} ref={ref} className={`${classNameProps} ${errorClass}`} {...inputProps} />
    </InputBox>
  )
})

InputControl.displayName = 'InputControl'
