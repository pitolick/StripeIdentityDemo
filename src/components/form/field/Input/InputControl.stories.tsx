import type { Meta, StoryObj } from '@storybook/react'
import { InputControl } from './InputControl'

const meta: Meta<typeof InputControl> = {
  title: 'Form/Field/Input/Control',
  component: InputControl,
  tags: ['autodocs'],
}

export default meta
type Story = StoryObj<typeof InputControl>

/**
 * 基本形
 */
export const Base: Story = {
  args: {
    type: 'text',
    nameId: 'base',
  },
}

/**
 * プレースホルダー
 */
export const Placeholder: Story = {
  args: {
    type: 'text',
    placeholder: '入力してください',
    nameId: 'placeholder',
  },
}

/**
 * 初期値
 */
export const InitialValue: Story = {
  args: {
    type: 'text',
    defaultValue: '初期値',
    nameId: 'initial-value',
  },
}

/**
 * ラベル
 */
export const Label: Story = {
  args: {
    type: 'text',
    label: 'ラベル',
    nameId: 'label',
  },
}

/**
 * 注釈
 */
export const Note: Story = {
  args: {
    type: 'text',
    note: { list: ['注釈'] },
    nameId: 'note',
  },
}

/**
 * バリデーション：必須
 */
export const Required: Story = {
  args: {
    type: 'text',
    label: '必須',
    required: true,
    nameId: 'required',
  },
}

/**
 * バリデーション：任意
 */
export const Optional: Story = {
  args: {
    type: 'text',
    label: '任意',
    required: false,
    nameId: 'optional',
  },
}

/**
 * バリデーション：エラー
 */
export const Error: Story = {
  args: {
    type: 'text',
    label: 'エラー',
    required: true,
    nameId: 'error',
    error: { type: 'required', message: '必須項目です' },
  },
}

/**
 * 基本形：テキスト
 */
export const Text: Story = {
  args: {
    type: 'text',
    label: 'テキスト',
    required: false,
    nameId: 'text',
    placeholder: '入力してください',
  },
}

/**
 * 基本形：パスワード
 */
export const Password: Story = {
  args: {
    type: 'password',
    label: 'パスワード',
    required: false,
    nameId: 'password',
    placeholder: 'パスワードを入力してください',
  },
}

/**
 * 基本形：日付
 */
export const Date: Story = {
  args: {
    type: 'date',
    label: '日付',
    required: false,
    nameId: 'date',
    placeholder: '日付を選択してください',
  },
}

/**
 * 基本形：時刻
 */
export const Time: Story = {
  args: {
    type: 'time',
    label: '時刻',
    required: false,
    nameId: 'time',
    placeholder: '時刻を選択してください',
  },
}

/**
 * 基本形：日時
 */
export const DateTime: Story = {
  args: {
    type: 'datetime-local',
    label: '日時',
    required: false,
    nameId: 'datetime',
    placeholder: '日時を選択してください',
  },
}

/**
 * 基本形：月
 */
export const Month: Story = {
  args: {
    type: 'month',
    label: '月',
    required: false,
    nameId: 'month',
    placeholder: '月を選択してください',
  },
}

/**
 * 基本形：週
 */
export const Week: Story = {
  args: {
    type: 'week',
    label: '週',
    required: false,
    nameId: 'week',
    placeholder: '週を選択してください',
  },
}

/**
 * 基本形：メールアドレス
 */
export const Email: Story = {
  args: {
    type: 'email',
    label: 'メールアドレス',
    required: false,
    nameId: 'email',
    placeholder: 'メールアドレスを入力してください',
  },
}

/**
 * 基本形：電話番号
 */
export const Tel: Story = {
  args: {
    type: 'tel',
    label: '電話番号',
    required: false,
    nameId: 'tel',
    placeholder: '電話番号を入力してください',
  },
}

/**
 * 基本形：URL
 */
export const Url: Story = {
  args: {
    type: 'url',
    label: 'URL',
    required: false,
    nameId: 'url',
    placeholder: 'URLを入力してください',
  },
}

/**
 * 基本形：検索
 */
export const Search: Story = {
  args: {
    type: 'search',
    label: '検索',
    required: false,
    nameId: 'search',
    placeholder: '検索してください',
  },
}

/**
 * 基本形：数値
 */
export const Number: Story = {
  args: {
    type: 'number',
    label: '数値',
    required: false,
    nameId: 'number',
    placeholder: '数値を入力してください',
  },
}

/**
 * 基本形：範囲
 */
export const Range: Story = {
  args: {
    type: 'range',
    label: '範囲',
    required: false,
    nameId: 'range',
    placeholder: '範囲を選択してください',
  },
}

/**
 * 基本形：ファイル
 */
export const File: Story = {
  args: {
    type: 'file',
    label: 'ファイル',
    required: false,
    nameId: 'file',
    placeholder: 'ファイルを選択してください',
  },
}

/**
 * 基本形：非表示
 */
export const Hidden: Story = {
  args: {
    type: 'hidden',
    nameId: 'hidden',
    placeholder: '非表示を選択してください',
  },
}
