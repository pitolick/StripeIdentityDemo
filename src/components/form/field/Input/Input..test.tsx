import React from 'react'
import { render } from '@testing-library/react'
import { Input } from './Input'

describe('Input コンポーネント', () => {
  it('正しくレンダリングされること', () => {
    const { container } = render(<Input />)
    expect(container).toMatchSnapshot()
  })

  it('classNameプロパティが指定された場合、正しく適用されること', () => {
    const { container } = render(<Input className="text-red" />)
    expect(container).toMatchSnapshot()
  })
})
