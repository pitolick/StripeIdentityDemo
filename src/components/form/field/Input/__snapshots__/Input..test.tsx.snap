// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input コンポーネント classNameプロパティが指定された場合、正しく適用されること 1`] = `
<div>
  <input
    class="text-red block w-full rounded-md border border-gray-500 px-3 py-2 text-sm file:mr-4 file:border-0 file:bg-gray-300 file:px-4 file:py-3 md:px-5 md:py-3 md:text-base"
  />
</div>
`;

exports[`Input コンポーネント 正しくレンダリングされること 1`] = `
<div>
  <input
    class=" block w-full rounded-md border border-gray-500 px-3 py-2 text-sm file:mr-4 file:border-0 file:bg-gray-300 file:px-4 file:py-3 md:px-5 md:py-3 md:text-base"
  />
</div>
`;
